### PR TITLE
Add percentage information to summary logger

### PR DESCRIPTION
**Pull request by natsegal**

### CODE REVIEW

File: nmtwizard/preprocess/consumer.py [hunk 1]
```diff
@@ -137,10 +141,25 @@ def _log_info(file_info, file_summary, proc, units):
             summary = self._summary[proc]
             if summary:
                 all_files_summary = summary.pop(None)
-                _log_info("for the whole sample ", all_files_summary, proc, units)
+                total_sentence_number = sum(
+                    [v.get("linesampled") for v in self._sampler_summary.values()]
```
Comment (guillaumekln) [inline nmtwizard/preprocess/consumer.py:145]: No need to build the list to compute the sum:

```suggestion
                    v.get("linesampled") for v in self._sampler_summary.values()
```

File: nmtwizard/preprocess/consumer.py [hunk 2]
```diff
@@ -90,7 +90,8 @@ def finalize(self):
 class SummaryLogger(Consumer):
     """A consumer that reduces operators filter information and logs the result."""
 
-    def __init__(self):
+    def __init__(self, sampler_summary):
+        self._sampler_summary = sampler_summary
         super().__init__()
```
Comment (guillaumekln) [inline nmtwizard/preprocess/consumer.py:95]: Prefer initializing the attribute after calling the parent constructor:

```suggestion
        super().__init__()
        self._sampler_summary = sampler_summary
```

File: nmtwizard/preprocess/consumer.py (hunk 1 above)
Comment (guillaumekln) [inline nmtwizard/preprocess/consumer.py:145]: Also, can the field "linesampled" be missing?
* If yes, it should be `v.get("linesampled", 0)`
* If not, it should be `v["linesampled"]`

File: nmtwizard/preprocess/consumer.py [hunk 3]
```diff
@@ -137,10 +141,25 @@ def _log_info(file_info, file_summary, proc, units):
...
+                )
                 sorted_by_base_name = sorted(summary.items(), key=lambda item: item[0])
                 for base_name, file_summary in sorted_by_base_name:
-                    _log_info(f"for the file '{base_name}' ", file_summary, proc, units)
+                    _log_info(
+                        f"for the file '{base_name}' ",
+                        file_summary,
+                        proc,
+                        units,
+                        self._sampler_summary.get(base_name, {}).get("linesampled"),
```
Comment (guillaumekln) [inline nmtwizard/preprocess/consumer.py:161]: Same comment regarding the "linesampled" field.

**Review by guillaumekln — CHANGES_REQUESTED** [review]

File: nmtwizard/preprocess/consumer.py (hunk 3 above)
Comment (natsegal) [reply nmtwizard/preprocess/consumer.py]: I think "linesampled" field is always present, same goes for the file name. So I replaced get with a simple lookup.

**Review by guillaumekln — APPROVED** [review]

**Pull request merged**